tests/contrib/actions/email/test_email_action.py

```diff
@@ -0,0 +1,195 @@
+import pytest
+
+from spidermon.contrib.actions.email import SendEmail
+from spidermon.exceptions import NotConfigured
+
+SENDER = "sender@example.com"
+RECIPIENT = "recipient@example.com"
+SUBJECT = "Subject"
+MESSAGE = "Message"
+BODY_TEXT = "Body Text"
+BODY_HTML = "Body HTML"
+
+
+@pytest.fixture
+def email(mocker):
+    email = SendEmail(
+        sender=SENDER,
+        to=RECIPIENT,
+        fake=True,
+        subject=SUBJECT,
+        body_text=BODY_TEXT,
+        body_html=BODY_HTML,
+    )
+    email.send_message = mocker.MagicMock()
+    mocked_message = mocker.MagicMock()
+    mocked_message.as_string = mocker.MagicMock()
+    email.get_message = mocker.MagicMock(return_value=mocked_message)
+    email.render_text_template = mocker.MagicMock()
+    email.render_template = mocker.MagicMock()
+    return email
+
+
+@pytest.mark.parametrize(
+    "sender,recipient,fake,subject,exception_message",
+    [
+        (
+            None,
+            None,
+            None,
+            None,
+            "You must provide at least one recipient for the message.",
+        ),
+        (SENDER, RECIPIENT, None, None, "You must provide a subject for the message."),
```

Comment:
What about having a `SENDER` but no `RECIPIENT`? I don’t see that in either `test_init_with_invalid_params` or `test_init_with_valid_params`.

Reply:
Thanks for the suggestion. It seems that I missed that test.